keps/sig-network/2681-pod-host-ip/README.md

```diff
@@ -284,11 +284,13 @@ The field is only informative, it doesn't affect running workloads.
 
 ###### What specific metrics should inform a rollback?
 
-The `status.hostIPs` field in Pod is empty, or frequently updated, or cause any other to crash.
+It will immediately update all running Pods on the node where this feature is enabled.
+
+If any of these phenomena imply that the feature is abnormal and needs to be rolled back, the `status.hostIPs` field in the pod is empty, or it is updated frequently, or it causes other Pods to crash.
 
 ###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
 
-TBD.
+It will test upgrade and rollback in e2e tests.
```

Comment:
Build and run the latest version using Kind

```
$ kind build node-image
$ kind create cluster --image kindest/node:latest
...
$ kubectl get node
NAME                 STATUS   ROLES           AGE     VERSION
kind-control-plane   Ready    control-plane   6m40s   v1.28.0-alpha.2.1529+c649dadff44981

```


Deploy a webserver
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: agnhost-server
  labels:
    app: agnhost-server
spec:
  replicas: 1
  selector:
    matchLabels:
      app: agnhost-server
  template:
    metadata:
      labels:
        app: agnhost-server
    spec:
      containers:
      - name: agnhost
        image: registry.k8s.io/e2e-test-images/agnhost:2.40
        args:
        - serve-hostname
        - --port=80
        ports:
        - containerPort: 80
```

Waiting pod be ready
```
$ kubectl get pod
NAME                                         READY   STATUS    RESTARTS   AGE
agnhost-server-76fb5c696c-2rqnh              1/1     Running   0          6s
```

Check pod hostIPs
```
$ kubectl get pod agnhost-server-76fb5c696c-2rqnh -o jsonpath='{.status.hostIPs}'
[{"ip":"172.18.0.2"}]
```

To disable the feature
```
$ docker exec -it kind-control-plane bash

$ cat <<EOF >>/var/lib/kubelet/config.yaml
featureGates:
  PodHostIPs: false
EOF

$ systemctl restart kubelet
```

Add more pod
```
$ kubectl scale --replicas=2 kubectl deploy/agnhost-server
```

Check that both Pods do not have HostIPs

```
$ kubectl get pod -o jsonpath='{.items[*].status.hostIPs}'
```

To enable the feature
```
$ docker exec -it kind-control-plane bash

$ sed -i 's/PodHostIPs: false/PodHostIPs: true/g' /var/lib/kubelet/config.yaml

$ systemctl restart kubelet
```

Check that both Pods have HostIPs
```
$ kubectl get pod -o jsonpath='{.items[*].status.hostIPs}'
[{"ip":"172.18.0.2"}] [{"ip":"172.18.0.2"}]
```

Reply:
I've manually verified that it looks like it works as expected, I'll try to see if it integrates with e2e_node first, and if it doesn't then I'll update this document.

Reply:
The above is definitely enough for me - if you can make it integrate with e2e_node that would be amazing, but just adding the above description to the doc would satisfy me :)